tfx/components/example_gen/utils_test.py

```diff
@@ -427,8 +518,66 @@ def testCalculateSplitsFingerprintSpanAndVersion(self):
-    self.assertEqual(span, '02')
-    self.assertEqual(version, '02')
+    self.assertEqual(span, 2)
+    self.assertEqual(version, 2)
+
+  def testCalculateSplitsFingerprintSpanAndVersionWithDate(self):
+    # Test align of span and version numbers.
+    span1_v1_split1 = os.path.join(self._input_base_path, '19700102', 'ver01',
+                                   'split1', 'data')
+    io_utils.write_string_file(span1_v1_split1, 'testing11')
+    span1_v1_split2 = os.path.join(self._input_base_path, '19700102', 'ver01',
+                                   'split2', 'data')
+    io_utils.write_string_file(span1_v1_split2, 'testing12')
+    span2_v1_split1 = os.path.join(self._input_base_path, '19700103', 'ver01',
+                                   'split1', 'data')
+    io_utils.write_string_file(span2_v1_split1, 'testing21')
+
+    # Test if error raised when date does not align.
+    splits = [
+        example_gen_pb2.Input.Split(
+            name='s1', pattern='{YYYY}{MM}{DD}/ver{VERSION}/split1/*'),
+        example_gen_pb2.Input.Split(
+            name='s2', pattern='{YYYY}{MM}{DD}/ver{VERSION}/split2/*')
+    ]
+    with self.assertRaisesRegexp(
+        ValueError, 'Latest span should be the same for each split'):
+      utils.calculate_splits_fingerprint_span_and_version(
+          self._input_base_path, splits)
+
+    span2_v1_split2 = os.path.join(self._input_base_path, '19700103', 'ver01',
+                                   'split2', 'data')
+    io_utils.write_string_file(span2_v1_split2, 'testing22')
+    span2_v2_split1 = os.path.join(self._input_base_path, '19700103', 'ver02',
+                                   'split1', 'data')
+    io_utils.write_string_file(span2_v2_split1, 'testing21')
+
+    # Test if error raised when date aligns but version does not.
+    splits = [
+        example_gen_pb2.Input.Split(
+            name='s1', pattern='{YYYY}{MM}{DD}/ver{VERSION}/split1/*'),
+        example_gen_pb2.Input.Split(
+            name='s2', pattern='{YYYY}{MM}{DD}/ver{VERSION}/split2/*')
+    ]
+    with self.assertRaisesRegexp(
+        ValueError, 'Latest version should be the same for each split'):
+      utils.calculate_splits_fingerprint_span_and_version(
+          self._input_base_path, splits)
+    span2_v2_split2 = os.path.join(self._input_base_path, '19700103', 'ver02',
+                                   'split2', 'data')
+    io_utils.write_string_file(span2_v2_split2, 'testing22')
+
+    # Test if latest span and version is selected when aligned for each split.
+    splits = [
+        example_gen_pb2.Input.Split(
+            name='s1', pattern='{YYYY}{MM}{DD}/ver{VERSION}/split1/*'),
+        example_gen_pb2.Input.Split(
+            name='s2', pattern='{YYYY}{MM}{DD}/ver{VERSION}/split2/*')
+    ]
+    _, span, version = utils.calculate_splits_fingerprint_span_and_version(
+        self._input_base_path, splits)
+    self.assertEqual(span, 2)
+    self.assertEqual(version, 2)
```

Comment:
is there a test that verify the split pattern updates of calculate_splits_fingerprint_span_and_version?

we want to make sure it's covered (especially for format like '02')

Reply:
added tests to check for updated split patterns